### PR TITLE
[Bot] Implement from issue #111: Create simple math utility with basic operations

### LINKED ISSUE #111: Create simple math utility with basic operations
﻿## Project Tag
project: math

## Description
Create a lightweight math utility module that provides basic arithmetic operations for use in other projects.

## Acceptance Criteria
- Function to add two numbers
- Function to multiply two numbers  
- Basic input validation
- Simple test to verify functions work
- Documentation with usage example

## Files to create
- \`math_utils.py\` - main utility functions
- \`test_math_utils.py\` - basic tests
- \`README.md\` - usage documentation

## Dependencies
- No external dependencies
- Use Python standard library only

## Scope & Segmentation Guidelines
**IMPORTANT FOR ANALYZER**: This is a simple, focused project. 
- Create maximum 3-4 logical tasks
- Group related activities together (don't create separate tasks for each function)
- Suggested segmentation: (1) Core implementation, (2) Testing, (3) Documentation
- Avoid micro-tasks like 'create __init__.py' or 'add single function'
- Keep tasks substantial and meaningful

This is a workflow verification test - prioritize logical task grouping over granular breakdown."

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #111

Project root: `projects`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-14 01:35:43 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of the math utility is functional but requires improvements in error handling and documentation. The test cases are a good start but can be enhanced for better clarity.

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math_utils.py:10`**: The functions do not handle edge cases such as adding or multiplying with None.
  *Suggestion*: Add checks to handle None inputs and raise appropriate exceptions.

#### 💡 SUGGESTION
- **`projects/test_math_utils.py:8`**: The test cases could be improved by using assertRaises context manager.
  *Suggestion*: Use 'with self.assertRaises(ValueError):' to improve readability.
- **`projects/README.md`**: The README could include examples for the multiply function.
  *Suggestion*: Add usage examples for the multiply function in the README.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math_utils.py
+++ b/projects/math_utils.py
@@ -5,6 +5,10 @@
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
     return a + b
+
+    if a is None or b is None:
+        raise ValueError("Arguments cannot be None.")
+
 def multiply(a, b):
     """Return the product of two numbers."""
     if not (isinstance(a, (int, float)) and isinstance(b, (int, float))):
         raise ValueError("Both arguments must be numbers.")
@@ -12,6 +16,10 @@
     return a * b
+
+    if a is None or b is None:
+        raise ValueError("Arguments cannot be None.")
```

**Patch 2**

```diff
--- a/projects/test_math_utils.py
+++ b/projects/test_math_utils.py
@@ -7,7 +7,8 @@
         self.assertRaises(ValueError, add, 2, "three")
+
+    def test_multiply(self):
         self.assertEqual(multiply(2, 3), 6)
-        self.assertRaises(ValueError, multiply, 2, "three")
+        with self.assertRaises(ValueError):
+            multiply(2, "three")
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!

**Comment by morotrade** [comment]
🔎 Using project root: `projects` for PR-fix

**Comment by morotrade** [comment]
❌ Failed to apply LLM patch in PR-fix mode. See logs.
